muschema/src/struct.ts

```diff
@@ -379,6 +381,41 @@ export class MuStruct<StructSpec extends _SchemaDictionary>
         });
         methods.patchBinary.push(`return result`);
 
+        // getByteLength subroutine
+        methods.getByteLength.push('var result=0;');
+        let byteLength = 0;
+        structProps.forEach((propName, i) => {
+            const type = structTypes[i];
+            switch (type.muType) {
+                case 'boolean':
+                case 'int8':
+                case 'uint8':
+                    ++byteLength;
+                    break;
+                case 'int16':
+                case 'uint16':
+                    byteLength += 2;
+                    break;
+                case 'float32':
+                case 'int32':
+                case 'uint32':
+                    byteLength += 4;
+                    break;
+                case 'float64':
+                    byteLength += 8;
+                    break;
+                case 'vector':
+                    byteLength += type.identity.byteLength;
+                    break;
+                case 'string':
+                    methods.getByteLength.push(`result+=4+Buffer.byteLength(x["${propName}"]);`);
```

Comment:
Buffer may not be available in a browser environment.  Maybe we could just use `x["${propName}"].length * 4` as a crude upper bound?

Reply:
yes, will do. didn't occur to me